lib/rucio/daemons/reaper/dark_reaper.py

```diff
@@ -178,47 +181,62 @@ def stop(signum=None, frame=None):
 
 
 def run(total_workers=1, chunk_size=100, once=False, rses=[], scheme=None,
-        exclude_rses=None, include_rses=None, delay_seconds=0, all_rses=False):
+        exclude_rses=None, include_rses=None, vos=None, delay_seconds=0):
     """
     Starts up the reaper threads.
 
     :param total_workers: The total number of workers.
     :param chunk_size: the size of chunk for deletion.
-    :param threads_per_worker: Total number of threads created by each worker.
     :param once: If True, only runs one iteration of the main loop.
-    :param greedy: If True, delete right away replicas with tombstone.
     :param rses: List of RSEs the reaper should work against. If empty, it considers all RSEs (Single-VO only).
     :param scheme: Force the reaper to use a particular protocol/scheme, e.g., mock.
     :param exclude_rses: RSE expression to exclude RSEs from the Reaper.
     :param include_rses: RSE expression to include RSEs.
+    :param vos: VOs on which to look for RSEs. Only used in multi-VO mode.
+                If None, we either use all VOs if run from "def", or the current VO otherwise.
     """
     logging.info('main: starting processes')
 
-    all_rses = list_rses()
```

Comment:
@bari12 just to be clear, when putting the option for `--all-rses` back in for the light/dark reaper I should also revert to this logic of listing the RSEs and overwriting the Boolean value of `all_rses` with a list (meaning it will always behave as though `--all-rses` was passed)?
I agree with your comment about making it a patch followed by feature, I just want to make sure I've understood what needs to be changed for this PR.

Reply:
Yes, I think so. We just keep it as it was and address the change of options in a feature.